scripts/run_fcst.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+"""
+The run script for run_fcst.
+"""
+
+import datetime as dt
+import logging
+import os
+import sys
+from argparse import ArgumentParser
+from copy import deepcopy
+from pathlib import Path
+
+from uwtools.api.config import get_yaml_config
+from uwtools.api.fs import link as uwlink
+from uwtools.api.fv3 import FV3
+from uwtools.api.logging import use_uwtools_logger
+from uwtools.api.template import render
+from uwtools.api.upp import UPP
+
+
+def _walk_key_path(config, key_path):
+    """
+    Navigate to the sub-config at the end of the path of given keys.
+    """
+    keys = []
+    pathstr = "<unknown>"
+    for key in key_path:
+        keys.append(key)
+        pathstr = " -> ".join(keys)
+        try:
+            subconfig = config[key]
+        except KeyError:
+            logging.error(f"Bad config path: {pathstr}")
+            raise
+        if not isinstance(subconfig, dict):
+            logging.error(f"Value at {pathstr} must be a dictionary")
+            sys.exit(1)
+        config = subconfig
+    return config
+
+
+def link_files(dest_dir, files):
+    """
+    Link a given list of files to the destination directory using the same file names.
+    """
+    for fpath in files:
+        path = Path(fpath)
+        linkname = dest_dir / path.name
+        if linkname.is_symlink():
+            linkname.unlink()
+        logging.info(f"Linking {linkname} -> {path}")
+        linkname.symlink_to(path)
+
+
+def parse_args(argv):
+    """
+    Parse arguments for the script.
+    """
+    parser = ArgumentParser(
+        description="Script that runs FV3 via uwtools API.",
+    )
+    parser.add_argument(
+        "-c",
+        "--config-file",
+        metavar="PATH",
+        required=True,
+        help="Path to experiment config file.",
+        type=Path,
+    )
+    parser.add_argument(
+        "--cycle",
+        help="The cycle in ISO8601 format (e.g. 2024-07-15T18).",
+        required=True,
+        type=dt.datetime.fromisoformat,
+    )
+    parser.add_argument(
+        "--key-path",
+        help="Dot-separated path of keys leading through the config to the driver's YAML block.",
+        metavar="KEY[.KEY...]",
+        required=True,
+        type=lambda s: s.split("."),
+    )
+    parser.add_argument(
+        "--member",
+        default="000",
+        help="The 3-digit ensemble member number.",
+    )
+    return parser.parse_args(argv)
```

Comment:
I'd imagine that developers would mostly copy existing patterns. If the pattern they observe is copy-paste, we'll certainly get more of that -- and then subtle drift over time as ad hoc changes are made without regard to the larger design. If they see much shorter code that uses something `import`ed, I suppose they'll copy that. I feel like we can at least try to set a good example and establish a useful pattern. Whether or not what I'm suggesting is "good" or "useful" -- and I agree that there's a balance between explicit and DRY -- is your call.

> I doubt this would be an enforceable construct to the community.
That's what PRs are for, but if reviewers won't request adherence to convention, there's not much we can do.

Reply:
> but if reviewers won't request adherence to convention, there's not much we can do.

That's been my main concern.

We get such a mix of developers in the apps that some will go for copy/paste within SRW, and others will try to write something from scratch (that's sometimes quite thrilling) or copy/paste it from somewhere else to try to slot it in. That doesn't work yet (give us time)! 

We could definitely try it out as a potential clean-up item.